### ISSUE TITLE
VRO code handoff: Review documentation of expiration/rotation/renewal of certs, secrets, etc.

### ISSUE THREAD

**Issue by dianagriffin** [department-of-veterans-affairs/abd-vro#1173]
**User Story**

As a VRO engineer, I want to confirm that documentation sufficiently covers expiration/rotation/renewal of certs, secrets, etc., so that I and my teammates can appropriately maintain them going forward.

Confirm that documentation identifies:
- expiration dates of certs, secrets, etc.
- playbooks with instructions on how to rotate/renew them

Review the documentation (see comments on this ticket) and update as necessary to cover these points as necessary. 

**Acceptance Criteria**

1. Documentation on expiration/rotation/renewal of certs, secrets, etc. has been reviewed and if necessary, updated, to identify the elements noted above with sufficient clarity.

**Not included in this work**

Actual management of certs, secrets, etc. is out of scope for this effort.

**Comment by dianagriffin** [comment]
Blocked while I work to get the relevant documentation from Amida

**Comment by pshahVA** [comment]
please let me know if you need me to work with Amida to unblock this.

**Comment by dianagriffin** [comment]
Requested updated docs specific to cert expiration/rotation [here](https://dsva.slack.com/archives/C01Q7979Z7D/p1682975305281879?thread_ts=1682359805.651329&cid=C01Q7979Z7D)

**Comment by dianagriffin** [comment]
From Afsin: 

> - Certificates issues are documented here https://github.com/department-of-veterans-affairs/abd-vro/wiki/BIP-APIs. 
> - prod-test certificate expires on January 6th, 2024 and production certificate expires on March 17th, 2024. 
> - The certificates are created using LHDi instructions here https://animated-carnival-57b3e7f5.pages.github.io/container-platform/cert-management/. 
>    - According to those instructions they will auto-renew about 4 months before the expiration date.  
> - We do have manual steps that are documented in our wiki to update Java truststore and keystore when they renew. Automating those manual steps will require more research and discussion with LHDI. 
> Let me know if you need any more information on the certificates.

**Comment by dianagriffin** [comment]
- Also need Lighthouse certs info from Amida
- Need BGS certs info from @nanotone

**Comment by nanotone** [comment]
The BGS microservice uses the exact same certs setup that the [BIP APIs do](https://github.com/department-of-veterans-affairs/abd-vro/wiki/BIP-APIs), and in fact the container takes the same three environment variables `$BIP_KEYSTORE`, `$BIP_TRUSTSTORE` and `$BIP_PASSWORD`.

**Comment by dianagriffin** [comment]
Follow up from Afsin: 

> the LH certificate in the secrets is a private key so there is no expiration date

**Comment by dianagriffin** [comment]
> The BGS microservice uses the exact same certs setup that the [BIP APIs do](https://github.com/department-of-veterans-affairs/abd-vro/wiki/BIP-APIs), and in fact the container takes the same three environment variables $BIP_KEYSTORE, $BIP_TRUSTSTORE and $BIP_PASSWORD.

Thanks @nanotone! Do you know the expiration dates of the BGS certs and how to rotate them?

**Comment by nanotone** [comment]
I could've been more direct in that previous comment: BGS microservice uses the exact same secrets that BIP API microservice uses (not just the same kind of secrets). So when expiration and rotation happens for BIP API microservice, it'll happen automatically for BGS microservice too with zero additional effort.

As for when and how, I don't know 🙃

**Comment by dianagriffin** [comment]
@nanotone ohhh, got it! thanks for clarifying -- I think we go the when/how from Afsin 👍

**Comment by dianagriffin** [comment]
Necessary documentation will be covered by #1163
